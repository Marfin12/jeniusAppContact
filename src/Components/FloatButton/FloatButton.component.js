import * as React from 'react';
import { Image } from 'react-native';
import Card from 'micro-card-marfin';

import { Images } from '../../Assets/Images';
import TextField from '../TextField';

import styles from './FloatButton.styles';
import config from './FloatButton.config';

const _renderAddButton = () => (
  <TextField textStyle={styles.iconPlus}>
    +
  </TextField>
);

const _renderDeleteButton = () =>
  <Image source={Images.delete_icon} style={styles.trashIcon} />

const FloatButton = (props) => (
  <Card style={[styles.floatButton, props.style]} onPress={props.onPress}>
    {props.addButton && _renderAddButton()}
    {props.deleteButton && _renderDeleteButton()}
  </Card>
);

FloatButton.displayName = config.displayName;
FloatButton.defaultProps = config.defaultProps;

export default FloatButton;
