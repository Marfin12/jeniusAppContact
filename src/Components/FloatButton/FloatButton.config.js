const defaultProps = {
  style: {},
  addButton: false,
  deleteButton: false,
  onPress: () => {}
};

const displayName = 'FloatButton';

export default {
  defaultProps,
  displayName,
};
