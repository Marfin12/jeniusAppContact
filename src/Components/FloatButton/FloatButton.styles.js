const styles = {
  floatButton: {
    alignItems: 'center',
    justifyContent: 'center',
    width: 60,  
    height: 60,
    borderRadius: 30,            
    backgroundColor: '#333333',                                    
    position: 'absolute',                                          
    top: 650,                                                    
    right: 10, 
  },
  trashIcon: {
    width: 36,
    height: 36
  },
  iconPlus: {
    fontSize: 40,
    color: '#ffffff'
  },
};

export default styles;
