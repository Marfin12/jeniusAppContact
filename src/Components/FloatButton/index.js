export { default } from './FloatButton.component';
