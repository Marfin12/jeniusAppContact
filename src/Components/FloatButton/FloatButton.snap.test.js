import React from 'react';
import renderer from 'react-test-renderer';

import FloatButton from './FloatButton.component';
import config from './FloatButton.config';

jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux');
  const lightMode = {
    theme: false,
  };
  const darkMode = {
    theme: true,
  };
  return {
    ...ActualReactRedux,
    useSelector: jest
      .fn()
      .mockImplementationOnce(() => lightMode)
      .mockImplementationOnce(() => darkMode),
  };
});
jest.mock('../../Components/TextField', () => 'TextField');
jest.mock('../../Components/Card', () => 'Card');
jest.mock('../../Assets/Images', () => 'Images');

afterEach(() => {
  jest.resetAllMocks();
});

describe('Float button Component Default Theme', () => {
  const ComponentWrapper = renderer
    .create(<FloatButton {...config.defaultProps} />)
    .toJSON();
  
  console.log(ComponentWrapper)

  test('renders float button component light mode', () => {
    expect(ComponentWrapper).toMatchSnapshot();
  });
});

describe('float button Component Dark Theme', () => {
  const ComponentWrapper = renderer
    .create(<FloatButton {...config.defaultProps} />)
    .toJSON();

  test('renders float button component dark mode', () => {
    expect(ComponentWrapper).toMatchSnapshot();
  });
});
