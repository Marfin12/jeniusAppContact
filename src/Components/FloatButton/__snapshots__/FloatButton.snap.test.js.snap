// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Float button Component Default Theme renders float button component light mode 1`] = `
<Card
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#333333",
        "borderRadius": 30,
        "height": 60,
        "justifyContent": "center",
        "position": "absolute",
        "right": 10,
        "top": 650,
        "width": 60,
      },
      Object {},
    ]
  }
/>
`;

exports[`float button Component Dark Theme renders float button component dark mode 1`] = `
<Card
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#333333",
        "borderRadius": 30,
        "height": 60,
        "justifyContent": "center",
        "position": "absolute",
        "right": 10,
        "top": 650,
        "width": 60,
      },
      Object {},
    ]
  }
/>
`;
